spinoso-string/src/enc/ascii/eq.rs

```diff
@@ -0,0 +1,75 @@
+use alloc::vec::Vec;
+
+use super::AsciiString;
+
+impl PartialEq<Vec<u8>> for AsciiString {
```

Comment:
I suspect that we'll need some sort of PartialEq checking between the encoded types.

```ruby
a = "hello"
b = "hello".force_encoding(Encoding::ASCII)
a == b # true
```
^-- however, maybe this just makes sense to implement `Eq` on the String class, and just compare bytes? (or is that what `cmp` is doing?